### ISSUE TITLE
Deno panic when using invalid syntax `(obj.prop: Object)`

### ISSUE THREAD

**Issue by yourfriendoss** [denoland/deno#13986]
# Error
```
============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: windows x86_64
Version: 1.19.2
Args: ["C:\Users\luke\.deno\bin\deno.EXE", "lsp"]

thread 'main' panicked at 'internal error: entered unreachable code: invalid pattern: Expr(Member(MemberExpr { span: Span { lo: BytePos(1), hi: BytePos(15), ctxt: #0 }, obj: Ident(Ident { span: Span { lo: BytePos(1), hi: BytePos(7), ctxt: #0 }, sym: Atom('packet' type=inline), optional: false }), prop: Ident(Ident { span: Span { lo: BytePos(8), hi: BytePos(15), ctxt: #0 }, sym: Atom('players' type=inline), optional: false }) }))', C:\Users\runneradmin.cargo\registry\src\github.com-1ecc6299db9ec823\swc_ecma_parser-0.88.3\src\parser\expr.rs:1604:44
note: run with RUST_BACKTRACE=1 environment variable to display a backtrace
```
# Reliable reproduction
`(packet.players: Object)`

**Comment by dsherret** [comment]
@yourfriendoss this is a bug because it shouldn't panic, but just FYI that's not what "casting"/type assertions look like. Do `packet.players as Object`.

**Comment by dsherret** [comment]
This was fixed in deno 1.20. It no longer panics.
